plugin/trino-mysql/src/main/java/io/trino/plugin/mysql/MySqlClient.java

```diff
@@ -1182,9 +1186,15 @@ Long getRowCount(JdbcTableHandle table)
         {
             RemoteTableName remoteTableName = table.getRequiredNamedRelation().getRemoteTableName();
             return handle.createQuery("""
-                        SELECT TABLE_ROWS FROM INFORMATION_SCHEMA.TABLES
-                        WHERE TABLE_SCHEMA = :schema AND TABLE_NAME = :table_name
-                        AND TABLE_TYPE = 'BASE TABLE'
+                        SELECT max(row_count) FROM (
+                            (SELECT TABLE_ROWS AS row_count FROM INFORMATION_SCHEMA.TABLES
+                            WHERE TABLE_SCHEMA = :schema AND TABLE_NAME = :table_name
+                            AND TABLE_TYPE = 'BASE TABLE')
+                            UNION ALL
+                            (SELECT CARDINALITY AS row_count FROM INFORMATION_SCHEMA.STATISTICS
+                            WHERE TABLE_SCHEMA = :schema AND TABLE_NAME = :table_name
+                            AND CARDINALITY IS NOT NULL)
```

Comment:
I suppose it's never possible for cardinality of any index to be > actual row count?

Reply:
i would guess this is still an estimate